library/cc/engine_builder.h

```diff
@@ -20,12 +22,15 @@ class EngineBuilder {
   EngineBuilder& addLogLevel(LogLevel log_level);
   EngineBuilder& setOnEngineRunning(std::function<void()> closure);
 
+  EngineBuilder& addStatsSinks(const std::vector<std::string>& stat_sinks);
   EngineBuilder& addGrpcStatsDomain(const std::string& stats_domain);
   EngineBuilder& addConnectTimeoutSeconds(int connect_timeout_seconds);
   EngineBuilder& addDnsRefreshSeconds(int dns_refresh_seconds);
   EngineBuilder& addDnsFailureRefreshSeconds(int base, int max);
   EngineBuilder& addDnsQueryTimeoutSeconds(int dns_query_timeout_seconds);
+  EngineBuilder& addDnsMinRefreshSeconds(int dns_min_refresh_seconds);
```

Comment:
related to https://github.com/envoyproxy/envoy-mobile/pull/2601#discussion_r992520428 - mhm, we are adding `add` methods now 🤔 I think that we should just change the following methods to use `add...` prefix too then:

```
  EngineBuilder& setAppVersion(const std::string& app_version);
  EngineBuilder& setAppId(const std::string& app_id);
  EngineBuilder& setDeviceOs(const std::string& app_id);
  EngineBuilder& setStreamIdleTimeoutSeconds(int stream_idle_timeout_seconds);
  EngineBuilder& setPerTryIdleTimeoutSeconds(int per_try_idle_timeout_seconds);
```

does not have to be done as part of this PR.

Reply:
Thanks for pointing that out. 👍 And thank you for working on improving the state of things in here - much appreciated.

I do not have a strong opinion as to whether to use `add` or `set` but going to share a few thoughts with regard to what I think the most optimal state could be. Side note: while I do not have a strong opinion as to whether it should be `set` or `add`  I think that we should clean up the APIs so that the name of a given method stays consistent between engine builder across Kotlin/Java/c++.

The name should depend on the nature of a given method:
* whether it's adding something (add) or just overriding a default value (set)
* what are the effects of calling a given method multiple times (whether `x+1`th call overrides the effects of `x`th call). If `x+1`th  call always overrides the result of `x`th call then `set`, otherwise consider `add`.

Reply:
Good advice. Will do!